tests/unit/test_session_recording.py

```diff
@@ -0,0 +1,248 @@
+import os
```

Comment:
would be nice to get a few comments in the longer test cases since it's hard for me personally to understand what's being tested actually